crates/subspace-fraud-proof/src/domain_extrinsics_builder.rs

```diff
@@ -0,0 +1,187 @@
+//! This module defines a trait for building the domain extrinsics from the original
```

Comment:
We have too much domain specific details here. I think this deserves to be in `domains` so that we can use this from there. Maybe block preprocessor ?